libvirt/resource_libvirt_dir_pool.go

```diff
@@ -0,0 +1,240 @@
+	}
+
+	err = pool.Create(0)
+	if err != nil {
+		return fmt.Errorf("Error starting libvirt storage pool: %s", err)
+	}
+
+	err = pool.Refresh(0)
+	if err != nil {
+		return fmt.Errorf("Error refreshing libvirt storage pool: %s", err)
+	}
+
+	id, err := pool.GetUUIDString()
+	if err != nil {
+		return fmt.Errorf("Error retrieving libvirt pool id: %s", err)
+	}
+	d.SetId(id)
+
+	// make sure we record the id even if the rest of this gets interrupted
+	d.Partial(true)
+	d.Set("id", id)
+	d.SetPartial("id")
+	d.Partial(false)
+
+	log.Printf("[INFO] Pool ID: %s", d.Id())
+
+	if err := poolWaitForExists(client.libvirt, id); err != nil {
+		return err
+	}
+
+	return resourceLibvirtDirPoolRead(d, meta)
+}
+
+func resourceLibvirtDirPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Client)
+	virConn := client.libvirt
+	if virConn == nil {
+		return fmt.Errorf(LibVirtConIsNil)
+	}
+
+	pool, err := virConn.LookupStoragePoolByUUIDString(d.Id())
+	if pool == nil {
+		log.Printf("storage pool '%s' may have been deleted outside Terraform", d.Id())
+		d.SetId("")
+		return nil
+	}
+	defer pool.Free()
+
+	poolName, err := pool.GetName()
+	if err != nil {
+		return fmt.Errorf("error retrieving pool name: %s", err)
+	}
+	d.Set("name", poolName)
+
+	info, err := pool.GetInfo()
+	if err != nil {
+		return fmt.Errorf("error retrieving pool info: %s", err)
+	}
+	d.Set("capacity", info.Capacity)
+	d.Set("allocation", info.Allocation)
+	d.Set("available", info.Available)
+
+	poolDefXML, err := pool.GetXMLDesc(0)
+	if err != nil {
+		return fmt.Errorf("could not get XML description for pool %s: %s", poolName, err)
+	}
+
+	var poolDef libvirtxml.StoragePool
+	err = xml.Unmarshal([]byte(poolDefXML), &poolDef)
+	if err != nil {
+		return fmt.Errorf("could not get a pool definition from XML for %s: %s", poolDef.Name, err)
+	}
+
+	var poolPath string
+	if poolDef.Target != nil && poolDef.Target.Path != "" {
+		poolPath = poolDef.Target.Path
+	}
+
+	if poolPath == "" {
+		log.Printf("Pool %s has no path specified", poolName)
+	} else {
+		log.Printf("[DEBUG] Pool %s path: %s", poolName, poolPath)
+		d.Set("path", poolPath)
+	}
+
+	return nil
+}
+
+func resourceLibvirtDirPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Client)
+	if client.libvirt == nil {
+		return fmt.Errorf(LibVirtConIsNil)
+	}
+
+	return deletePool(client, d.Id())
+}
+
+func resourceLibvirtDirPoolExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	log.Printf("[DEBUG] Check if resource libvirt_dir_pool exists")
+	client := meta.(*Client)
+	virConn := client.libvirt
+	if virConn == nil {
+		return false, fmt.Errorf(LibVirtConIsNil)
+	}
+
+	pool, err := virConn.LookupStoragePoolByUUIDString(d.Id())
+	if err != nil {
+		virErr := err.(libvirt.Error)
+		if virErr.Code != libvirt.ERR_NO_STORAGE_POOL {
+			return false, fmt.Errorf("Can't retrieve pool %s", d.Id())
+		}
+		// does not exist, but no error
```

Comment:
can you give more info about this context?

Reply:
This is the case where you handle the manual deletion. The comment was C&P with the code from volume code.

Reply:
ok thx for clarification